src/renderer/screens/settings/sections/Help/ResetButton.js

```diff
@@ -22,17 +25,28 @@ export default function ResetButton() {
     { open, close, closeFallback, handleConfirm, handleError },
   ] = useActionModal();
 
-  const onConfirm = useCallback(async () => {
-    if (pending) return;
+  const globalSyncState = useGlobalSyncState();
+  const cvPolling = useCountervaluesPolling();
+  const isPending = cvPolling.pending || globalSyncState.pending;
+
+  reset = async () => {
+    if (isPending) {
+      return setTimeout(reset, 1000);
```

Comment:
setTimeout do not return a promise so i'm not sure what is going on here.

wouldn't it be simpler to have a "isPending()" predicate that you call in a

```
while(isPending()) {
  await delay(1000);
}
```

Reply:
i am not retuning a promise, i am just waiting to recall the function.
indeed i did not think to do it this way. (need to be careful with the isPending to be sure it uses an updated version of isPending in the logic though).